python/lsst/afw/image/exposure/exposureUtils.py

```diff
@@ -47,66 +49,53 @@ def bbox_to_convex_polygon(bbox, wcs, padding=10):
     convex_polygon : `lsst.sphgeom.ConvexPolygon`
        Will be None if wcs is not valid.
     """
-    if wcs is None:
-        return None
-
+    # Convert Box2I to Box2D, without modifying original.
     _bbox = lsst.geom.Box2D(bbox)
     _bbox.grow(padding)
     corners = [wcs.pixelToSky(corner).getVector()
                for corner in _bbox.getCorners()]
     return lsst.sphgeom.ConvexPolygon(corners)
 
 
-def bbox_contains_sky_coords(bbox, wcs, ra, dec, degrees=False, padding=10):
+def bbox_contains_sky_coords(bbox, wcs, ra, dec, padding=10):
     """Check if a set of sky positions are in the bounding box.
 
-    This does a robust two-step procedure to avoid inverting the WCS outside
-    of the valid region.
+    This uses a two-step process: first check that the coordinates are
+    inside a padded version of the bbox projected on the sky, and then
+    project the remaining points onto the bbox, to avoid inverting
+    the WCS outside of the valid region. The default padding
+    size was chosen to be sufficient for the most warped detectors at
+    the edges of the HyperSuprimeCam focal plane.
 
     Parameters
     ----------
     bbox : `lsst.geom.Box2I`
-        Bounding box to convert.
+        Pixel bounding box to check sky positions in.
     wcs : `lsst.afw.image.SkyWcs`
         WCS associated with the bounding box.
-    ra : `np.ndarray`
-        Array of Right Ascension.  Units are radians unless degrees=True.
-    dec : `np.ndarray`
-        Array of Declination.  Units are radians unless degrees=True.
-    degrees : `bool`, optional
-            Input ra, dec arrays are degrees if True.
+    ra : `astropy.Quantity`, (N,)
+        Array of Right Ascension, angular units.
+    dec : `astropy.Quantity`, (N,)
+        Array of Declination, angular units.
     padding : `int`
        Pixel padding to ensure that bounding box is entirely contained
        within the resulting polygon.
 
     Returns
     -------
-    contained : `np.ndarray`
-       Boolean indicating which points are contained in the bounding box.
-
-    Raises
-    ------
-    ValueError if exposure does not have a valid wcs.
+    contained : `np.ndarray`, (N,)
+       Boolean array indicating which points are contained in the
+       bounding box.
     """
     poly = bbox_to_convex_polygon(bbox, wcs, padding=padding)
 
-    if poly is None:
-        raise ValueError("Exposure does not have a valid wcs.")
-
-    if degrees:
-        _ra = np.deg2rad(np.atleast_1d(ra)).astype(np.float64)
-        _dec = np.deg2rad(np.atleast_1d(dec)).astype(np.float64)
-    else:
-        _ra = np.atleast_1d(ra).astype(np.float64)
-        _dec = np.atleast_1d(dec).astype(np.float64)
+    _ra = np.atleast_1d(ra.to(units.radian).value).astype(np.float64)
+    _dec = np.atleast_1d(dec.to(units.radian).value).astype(np.float64)
```

Comment:
You can do `ra.to_value(units.radian)`, though it looks like that doesn't result in a measureable speedup.